ocaml/src/proxy_protocol.ml

```diff
@@ -530,6 +530,7 @@ module Protocol = struct
       | NoSatisfiablePolicy     [@value 15]
       | ProtocolVersionMismatch [@value 17]
       | AssertFailed            [@value 18]
+      | BadArgument             [@value 19]
```

Comment:
This error type can only be returned as the result of an operation that the proxy doesn't expose, so it seems a bit weird to add this here?
(I would opt for handling Bad_argument in proxy_server.ml in the same way as e.g. Nsm_host_already_exists is already handled.)